### ISSUE TITLE
Question about the project

### ISSUE THREAD

**Issue by ArjixWasTaken** [rust-av/matroska#97]
Once this project is "complete" (well, the basic stuff are done), will I be able to use it to make an mkv file from an mp4 file?
A 100% rust solution for mkv would be awesome to have, cause it means I don't need any external deps.

**Comment by lu-zero** [comment]
The answer is yes. How soon depends on how many people get involved.

**Comment by ArjixWasTaken** [comment]
That's great!

I am interested in helping, but this would be my first time working on a binary format like this, could you share your workflow?
Like, how do you implement the rfc?

**Comment by Luni-4** [comment]
Just read about this issue, sorry the late reply

I don't know if there is a general workflow, I can talk about my experience when I started contributing to this project. I followed these steps:

1. Studied `nom` crate, browsing its repository and fiddling with its examples
2. Applied what I learnt about `nom` to this repository, trying to figure out how functions were structured
3. Looked at the `rfc` to understand the meaning behind each function and how the series of functions associated to a specific part of the rfc had been implemented
4. Fiddled with the code, making many and many mistakes

I hope this could answer to your questions!
